### PR TITLE
Refactor repository

### PR DESCRIPTION
**Pull request by geaaru**
Travis said that is ok. :)

### CODE REVIEW

File: pkg/installer/installer_test.go [hunk 1]
```diff
@@ -86,12 +86,14 @@ var _ = Describe("Installer", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(repo.GetName()).To(Equal("test"))
 			Expect(helpers.Exists(spec.Rel("repository.yaml"))).ToNot(BeTrue())
-			Expect(helpers.Exists(spec.Rel(TREE_TARBALL))).ToNot(BeTrue())
+			Expect(helpers.Exists(spec.Rel(TREE_TARBALL + ".gz"))).ToNot(BeTrue())
```
Comment (mudler) [inline pkg/installer/installer_test.go]: Does it means we don't test anymore the case when it is *not* compressed?

File: cmd/create-repo.go [hunk 2]
```diff
@@ -123,7 +141,9 @@ func init() {
 	createrepoCmd.Flags().String("repo", "", "Use repository defined in configuration.")
 
 	createrepoCmd.Flags().String("tree-compression", "none", "Compression alg: none, gzip")
-	createrepoCmd.Flags().String("tree-path", installer.TREE_TARBALL, "Repository tree filename")
+	createrepoCmd.Flags().String("tree-name", installer.TREE_TARBALL, "Repository tree filename")
```
Comment (mudler) [inline cmd/create-repo.go]: `tree-name` doesn't tell much about the file, but looks like the "tree name" .. maybe better `tree-filename` instead ?

**Comment by mudler** [comment]
This might need to be updated https://github.com/Luet-lab/docs/blob/364cdf0e7659cd2a617dbb9508ad53a16f277437/content/en/docs/Overview/repositories.md#flags

File: pkg/installer/installer_test.go (hunk 1 above)
Comment (geaaru) [reply pkg/installer/installer_test.go]: Yup, we need to add a test for this :D

File: cmd/create-repo.go (hunk 2 above)
Comment (geaaru) [reply cmd/create-repo.go]: so similar for `meta-filename`?

File: pkg/installer/installer_test.go (hunk 1 above)
Comment (geaaru) [reply pkg/installer/installer_test.go]: Done

**Pull request merged**